JSX.md

```diff
@@ -58,19 +54,25 @@ The `as` operator is available in both `.ts` and `.tsx` files.
 
 # Type Checking
 
-In order to understand type checking with JSX you must first understand intrinsic elements vs value-based elements. Given a JSX expression `<expr />`, is `expr` referring to something intrinsic to the environment (ie, a div or span in a DOM environment) or to a custom component that you've made? This is important for two reasons:
+In order to understand type checking with JSX you must first understand the difference between intrinsic elements value-based elements.
+Given a JSX expression `<expr />`, `expr` may either refer to something intrinsic to the environment (e.g. a `div` or `span` in a DOM environment) or to a custom component that you've created.
+This is important for two reasons:
 
-1. For React, intrinsic elements are emitted as strings, like `React.createElement('div')`, whereas a component you've created is not: `React.createElement(MyComponent)`.
+1. For React, intrinsic elements are emitted as strings (`React.createElement("div")`), whereas a component you've created is not (`React.createElement(MyComponent)`).
 2. The types of the attributes being passed in the JSX element should be looked up differently. Intrinsic element attributes should be known *intrinsically* whereas components will likely want to specify their own set of attributes.
```

Comment:
"just be known" was bad phrasing on my part. Yes, you still need to tell the type system about the attribute.

The point is that there are two types of JSX elements.. those that are declared to be known (similar to declaring a (external) module or variable) and those that you create. For the first kind, since the element itself is declared, it follows that the attributes of the element are also declared. For the second, it also follows that the attributes of the element are explicitly defined when you create the element.

It's somewhat analogous to this:

```typescript
// intrinsic.d.ts
declare module "intrinsic" {
  function intrinsic(foo: string);
  export = intrinsic;
}

// value.ts
function value(bar: boolean) { }
export = value;

// app.ts
import intrinsic = require('intrinsic');
import value = require('./value');

intrinsic('foobar');
value(true);
```

So verbatim rewording the original text of (2):

The types of the parameters being passed to the function exported by the module should be looked up differently. Parameters of the declared module should be known *based on the declaration file* whereas parameters of the defined module should be known based on the function definition.

Not a perfect translation but maybe helps get the point across. Also: I'm all for rewording that section to something else that makes more sense.